### ISSUE TITLE
More "Variations" In items.

### ISSUE THREAD

**Issue by DatPerkYT** [BEEmod/BEE2-items#3958]
### Description of the feature

I think that a few items should have more variations, for example, the Turret. we should have an option to make it defective, backwards, and so on. I think most of these should be the "Cube Type" Option. Here's a few items and what I think they should have:
1. Turrets being able to be defective.
2. Portal Gun being able to change on other styles, like Portal 1 and Mel.
3. Fizzlers being able to change to FourthReaper's Fizzlers, like the Compressed Smoke Field and so on.
4. A Bottomless pit option in the Goo, i can't find out how to change it! (Still by Cube Type!)
5. BEING ABLE TO GROUP ITEMS TOGETHER! This would be so helpful to improve space, by Middle clicking (Really any key idc) then using that key on another object and it would be Grouped together. (Just a cool idea me and my friend came up with)

### Why should this be implemented

I think doing this would make BEEmod So much more Customizable, and A TON better! Again, These are just good ideas that would make me (and probably the community) so much happier when using BEEmod! I never get attention on any post, so i hope i at least get a comment on this one. 

### Additional information

As i said, all of the options would be in "Cube Type", because there is a lot of options there. i have no pictures because i know NOTHING about coding whatsoever. Sorry :/

**Comment by Luke18033** [comment]
Please don't combine suggestions

**Comment by CardsAndGamesYT** [comment]
My brain hurts

**Comment by Luke18033** [comment]
- Defective turrets aren't being added, what puzzle purpose would they serve? See #3821 for backwards turrets.
- The Mel developers said they don't want other people using their portal gun model, and the P1 gun is already implemented.
- See #1596 for combining custom fizzlers.
- Bottomless pits are being reworked and aren't implemented currently.
- We can't combine random items, the fizzlers are really the only thing that could be combined but aren't. We hate the limited palette space as much as you do, we would have done something about it long ago if it were possible.

**Comment by DatPerkYT** [comment]
oh ok. i understand.

**Comment by DatPerkYT** [comment]
Sorry for combining it, iv'e never done this before. this is my first one.

**Comment by CardsAndGamesYT** [comment]
> - Defective turrets aren't being added, what puzzle purpose would they serve? See #3821 for backwards turrets.
> 
> - The Mel developers said they don't want other people using their portal gun model, and the P1 gun is already implemented.
> 
> - See #1596 for combining custom fizzlers.
> 
> - Bottomless pits are being reworked and aren't implemented currently.
> 
> - We can't combine random items, the fizzlers are really the only thing that could be combined but aren't. We hate the limited palette space as much as you do, we would have done something about it long ago if it were possible.

Also, Mel isn't even a style

**Comment by DatPerkYT** [comment]
CardsAndGamesYT i was just saying that if they coould, add the Portal Gun "Cube Type" and make the unused ones the Mel Portal gun and the Portal 1, even if the style was say Overgrown, the Portal 1 Gun would be it.

**Comment by DatPerkYT** [comment]
Guys, these are just ideas. I came up with them because I thought Defective turrets would be a cool way to make it  more "Wheatley" style, like he forgot to change them back after him and Chell made them all defective. I understand the Mel portal gun one, I did not know Combined Fizzlers existed, or that Bottomless pits were being reworked. and yes I also understand the last one it would have been too hard and stuff to implement.
